op-geth-proxy/geth-proxy.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+
+	"github.com/peterbourgon/ff/v3"
+)
+
+// Environment variables beginning with this prefix can be used to instantiate command line flags
+const ENV_PREFIX = "SEQUENCER_PROXY"
+
+var (
+	fs            = flag.NewFlagSet("proxy", flag.ContinueOnError)
+	fromAddr      = fs.String("listen-addr", "127.0.0.1:9090", "proxy's listening adress")
+	sequencerAddr = fs.String("seq-addr", "http://127.0.0.1:50000", "address of espresso sequencer")
+	gethAddr      = fs.String("geth-addr", "http://127.0.0.1:8545", "address of the op-geth node")
+	vm_id         = fs.Int("vm-id", 1, "VM ID of the OP rollup instance")
+)
+
+type Transaction struct {
+	Vm      int   `json:"vm"`
+	Payload []int `json:"payload"`
```

Comment:
Should this be like `char` or `int8` or something? Or is there a `bytes` type or something like that in Go?

Reply:
yes - there is a bytes type. This type serializes as a base 64 encoded [string](https://stackoverflow.com/questions/14177862/how-to-marshal-a-byte-uint8-array-as-json-array-in-go) (which our sequencer API fails to parse). I didn't really want to mess with writing a custom Marshaler, so to get around this I cast the unmarshalled []byte into an []int [here](https://github.com/EspressoSystems/op-espresso-integration/blob/proxy/op-geth-proxy/geth-proxy.go#L41).

Reply:
I see, fair enough